modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -1918,6 +1910,12 @@ protected void invalidated() {
                     pseudoClassStateChanged(DISABLED_PSEUDOCLASS_STATE, get());
                     updateCanReceiveFocus();
                     focusSetDirty(getScene());
+
+                    if (Node.this instanceof Parent parent) {
+                        for (Node child : parent.getChildren()) {
+                            child.updateDisabled();
+                        }
```

Comment:
If you like this style, you could do `parent.getChildren().forEach(Node::updateDisabled);`. I find these more readable, but it's up to you.

Reply:
Done.